### DISCUSSION TITLE
Is there a way to get a week's worth of data?

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#1892] [Questions] [answered]
I want to do a fileter with created_at data.
Is there a way to do this like `where created_at BETWEEN`?

**Comment by steve-chavez** [comment, marked as answer]
Try with

```sql
select ... 
where created_at between now() - '1 week'::interval and now()
```

**Reply by ru88s** [reply]
@steve-chavez, Thanks.

```
const { data, error, count } = await supabase
        .from("comments")
        .select("*", { count: "exact" })
        where created_at between now() - '1 week'::interval and now()
        .range(from, to);
```
<img width="479" alt="スクリーンショット 2021-06-10 7 55 24" src="https://user-images.githubusercontent.com/42270511/121439605-37fb0180-c9c1-11eb-99ec-f9ce1553a848.png">

How can I apply this to Filters?

**Reply by inian** [reply]
Try combining with the [gt](https://supabase.io/docs/reference/javascript/gt) and lt filters @ru88s? I haven't tested it but something like this should work
```
const { data, error, count } = await supabase
        .from("comments")
        .select("*", { count: "exact" })
        .gt('created_at', new Date(Date.now() - 604800000))
        .lt('created_at', Date.now())
        .range(from, to);
```

**Reply by ru88s** [reply]
@inian, Thanks! It's null now. What's the problem?

<img width="644" alt="スクリーンショット 2021-06-10 20 53 11" src="https://user-images.githubusercontent.com/42270511/121520798-50553580-ca2e-11eb-877e-51da05b09d19.png">

![スクリーンショット 2021-06-10 20 53 24](https://user-images.githubusercontent.com/42270511/121520805-521ef900-ca2e-11eb-8154-772635d97b23.png)

![スクリーンショット 2021-06-10 20 54 39](https://user-images.githubusercontent.com/42270511/121520806-52b78f80-ca2e-11eb-99c0-9a60ae6fdbdd.png)

**Reply by inian** [reply]
Hey @ru88s, my suggestion was wrong. Try

```js
const { data, error, count } = await supabase
        .from("comments")
        .select("*", { count: "exact" })
        .gt('created_at', new Date(Date.now() - 604800000).toISOString())
        .lt('created_at', new Date().toISOString() )
        .range(from, to);
```

**Reply by ru88s** [reply]
Oh! @inian, I'm sorry it took me so long to notice!
It worked! Thank you very much!
